### DISCUSSION TITLE
Welcome to unirest-java Discussions!

### DISCUSSION THREAD

**Discussion by ryber** [Kong/unirest-java#390] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by RobLewis** [comment]
Hi. Java & Android hobbyist here. Just discovered Unirest and it looks impressive! 
Poking around I've seen a couple of references to a v. 4 (for example, WebSocket support). Is there a formal development effort underway, design document, milestones, schedule, anything like that?

**Reply by ryber** [reply]
Hi @RobLewis, yes, WebSocket support is being built into the RC branch https://github.com/Kong/unirest-java/tree/jv11 however the WebSicket part is not exposed as of yet. It will be there by RC2. I suspect if the weather continues to be horrible I will probably get to it soon

It will require Java 11. I don't have a solid timeline on exactly WHEN it will be released as "non-RC". It would be great to have some beta testers.

**Reply by RobLewis** [reply]
I'm not exactly a real HTTP stud but I'd give it a go.

**Comment by nenadmilasinovic** [comment]
Hi, 
I am facing a problem to send file in POST request in body.
How can i simulate Postman call with form-data body, key=file and value=exact file.
```
curl --location --request POST 'https://api.testing.rs/api/org/v1/invoice/send/file?pibSender=105622807&pibReceiver=106946734&contactName=test&contactEmail=test@gmail.com' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer token' \
--form 'file=@"someFile.pdf"'

```
I tried with following without success:

```
CompletableFuture<HttpResponse<SendInvoiceResponse>> future = Unirest.post("https://api.testing.rs/api/org/v1/invoice/send/file")
                        .header("accept", "application/json")
                        .header("Content-Type", "application/pdf")
                        .header("Authorization", "Bearer " + token)
                        .queryString("senderPib", "105622807")
                        .queryString("receiverPib", "106946734")
                        .queryString("contactName", "test")
                        .queryString("contactEmail", "test@gmail.com")
                        .field("file", new File(filePath))
                    .asObjectAsync(SendInvoiceResponse.class, new Callback<SendInvoiceResponse>() {      
                       public void failed(UnirestException e) {
                           logger.info("The request has failed");
                       }
     
                       public void completed(HttpResponse<SendInvoiceResponse> response) {
                           int code = response.getStatus();
                           Headers headers = response.getHeaders();
                           SendInvoiceResponse body = response.getBody();
                       }
     
                       public void cancelled() {
                           logger.info("The request has been cancelled");
                       }
                });

```
As a result i get code: failed, file is missing.

Thank you in advance.

**Reply by ryber** [reply]
are you getting that from the remote server or locally? Is it an IO exception from java? Is maybe the file path wrong?

**Reply by ryber** [reply]
@iceDiceTD one weird thing that might be throwing things off. You have listed the Content-Type as "application/json", but that is not what you are sending. You are sending a multi-part form with a single field. If you leave off the Content-Type Unirest will generate the proper header

**Comment by elizarrrrr** [comment]
Hi guys, thanks for the library you made, looks really clean.

But I got a question.

I am trying to retrieve data from jira. The response object has following format.

{
   "size": 1,
    "start": 0,
    "limit": 50,
    "values": [
    list of requests.....
    ]
}

Now how can I access this list? Because when I do 
```
.AsObect(new GenericType<List<Book>>() {
                }  

```
I get the following error:
![grafik](https://user-images.githubusercontent.com/36076242/158343638-fdc321c6-e6c4-432d-aad8-dd7e4ebb1182.png)
Looking at this, I would think that i need an object where values is an attribute in it to retrieve the data, but there is for sure a better way, isn't it?
